apps/aecontract/src/aect_create_tx.erl

```diff
@@ -128,13 +128,26 @@ nonce(#contract_create_tx{nonce = Nonce}) ->
 origin(#contract_create_tx{owner = OwnerPubKey}) ->
     OwnerPubKey.
 
-%% Owner should exist, and have enough funds for the fee
--spec check(tx(), aetx:tx_context(), aec_trees:trees(), height(), non_neg_integer()) -> {ok, aec_trees:trees()} | {error, term()}.
-check(#contract_create_tx{owner = OwnerPubKey, nonce = Nonce,
+%% Owner should exist, and have enough funds for the fee, the amount and the gas
+-spec check(tx(), aetx:tx_context(), aec_trees:trees(), height(), non_neg_integer()) ->
+                   {ok, aec_trees:trees()} | {error, term()}.
+check(#contract_create_tx{owner = OwnerPubKey,
+                          nonce = Nonce,
+                          amount     = Amount,
+                          gas        = Gas,
+                          gas_price  = GasPrice,
                           fee = Fee}, _Context, Trees, Height, _ConsensusVersion) ->
+    TotalAmount = Fee + Amount + Gas * GasPrice,
     Checks =
-        [fun() -> aetx_utils:check_account(OwnerPubKey, Trees, Height, Nonce, Fee) end],
-
+        [fun() -> aetx_utils:check_account(OwnerPubKey, Trees, Height, Nonce, TotalAmount) end,
```

Comment:
This checks that originator of tx is in accounts state tree with enough balance.
I would expect (sanity) check that computed contract pubkey is absent from account state tree.

Reply:
There is nothing stopping you from transferring funds to a contract address before it exists (you can calculate the address beforehand) thus creating the address and adding it to the state tree. Hence I do not think we should check that the account doesn't exist.